api/src/main/java/org/apache/iceberg/expressions/LiteralSet.java

```diff
@@ -0,0 +1,212 @@
+    }
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public boolean equals(Object other) {
+    if (this == other) {
+      return true;
+    }
+    if (other == null || getClass() != other.getClass()) {
+      return false;
+    }
+    LiteralSet<T> that = (LiteralSet<T>) other;
+    return literals.equals(that.literals);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(literals);
+  }
+
+  @Override
+  public boolean contains(Object object) {
+    return literals.contains(object);
+  }
+
+  @Override
+  public int size() {
+    return literals.size();
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return literals.isEmpty();
+  }
+
+  @Override
+  public Iterator<T> iterator() {
+    return literals.iterator();
+  }
+
+  @Override
+  public Object[] toArray() {
+    return literals.toArray();
+  }
+
+  @Override
+  public <X> X[] toArray(X[] a) {
+    return literals.toArray(a);
+  }
+
+  @Override
+  public boolean containsAll(Collection<?> c) {
+    return literals.containsAll(c);
+  }
+
+  @Override
+  public boolean add(T t) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public boolean remove(Object o) {
+    throw new UnsupportedOperationException();
+  }
+
+
+  @Override
+  public boolean addAll(Collection<? extends T> c) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public boolean retainAll(Collection<?> c) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public boolean removeAll(Collection<?> c) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void clear() {
+    throw new UnsupportedOperationException();
+  }
+
+  static class CharSeqWrapper implements CharSequence, Serializable {
+    private static final Comparator<CharSequence> CMP =
+        Comparators.<CharSequence>nullsFirst().thenComparing(Comparators.charSequences());
+
+    private final CharSequence chars;
+
+    CharSeqWrapper(CharSequence charSequence) {
+      Preconditions.checkNotNull(charSequence, "String literal values cannot be null");
+      this.chars = charSequence;
+    }
+
+    @Override
+    public int length() {
+      return chars.length();
+    }
+
+    @Override
+    public char charAt(int index) {
+      return chars.charAt(index);
+    }
+
+    @Override
+    public CharSequence subSequence(int start, int end) {
+      return chars.subSequence(start, end);
+    }
+
+    @Override
+    public int hashCode() {
+      return chars.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (this == obj) {
+        return true;
+      }
+      if (obj == null) {
+        return false;
+      }
+
+      if (obj instanceof CharSequence) {
+        return CMP.compare(chars, (CharSequence) obj) == 0;
+      }
+      return false;
+    }
+
+    @Override
+    public String toString() {
+      return chars.toString();
+    }
+
+    public CharSequence unWrap() {
```

Comment:
Can you change this to `get` instead? Actually, I think this is already implemented on the other wrapper.

Reply:
👍  removed it completely.